nixos/modules/services/matrix/mautrix-meta.nix

```diff
@@ -0,0 +1,210 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.mautrix-meta;
+  fullDataDir = "/var/lib/${cfg.dataDir}";
+  relativeDataDir = cfg.dataDir;
+  pkg = cfg.package;
+  registrationFile = "${fullDataDir}/meta-registration.yaml";
+  settingsFormat = pkgs.formats.yaml {};
+  settingsFile = "${fullDataDir}/config.yaml";
+  settingsFileUnsubstituted =
+    settingsFormat.generate "mautrix-meta-config.yaml" cfg.settings;
+in {
+  options = {
+    services.mautrix-meta = {
+      enable = mkEnableOption (mdDoc "Mautrix-Meta, a Matrix <-> Facebook and Matrix <-> Instagram hybrid puppeting/relaybot bridge");
+
+      package = mkPackageOption pkgs "mautrix-meta" { };
+
+      dataDir = mkOption {
+        type = types.path;
+        default =  "mautrix-meta";
+        description = mdDoc ''
+          Path to the directory with database, registration, and other data for the bridge service.
+          This path is relative to `/var/lib`, it cannot start with `../` (it cannot be outside of `/var/lib`).
+        '';
+      };
+
+      settings = mkOption rec {
+        apply = recursiveUpdate default;
+        inherit (settingsFormat) type;
+        default = {
+          homeserver = {
+            software = "standard";
+          };
+
+          appservice = rec {
+            database = {
+              type = "sqlite3-fk-wal";
+              uri = "file:${fullDataDir}/mautrix-meta.db?_txlock=immediate";
+            };
+
+            hostname = "0.0.0.0";
+            port = 29319;
+            address = "http://localhost:${toString cfg.settings.appservice.port}";
+          };
+
+          meta = {
+            mode = "instagram";
+          };
+
+          logging = {
+            min_level = "info";
+            writers = singleton {
+              type = "stdout";
+              format = "pretty-colored";
+              time_format = " ";
+            };
+          };
```

Comment:
I wonder if we should configure encryption by default. It is getting a bit into the opinionated zone but I like encouraging secure options.

https://github.com/NixOS/nixpkgs/blob/8ab9a91a1a26752572d2147933caf8ab6faa5f3f/nixos/modules/services/matrix/mautrix-facebook.nix#L46-L55

Reply:
What do you think about these options:
```
                # Enable encryption by default to make the bridge more secure
                encryption = {
                  allow = true;
                  default = true;

                  # Recommended options from mautrix documentation
                  # for additional security.
                  delete_keys = {
                    dont_store_outbound = true;
                    ratchet_on_decrypt = true;
                    delete_fully_used_on_decrypt = true;
                    delete_prev_on_new_session = true;
                    delete_on_device_delete = true;
                    periodically_delete_expired = true;
                    delete_outdated_inbound = true;
                  };

                  verification_levels = {
                    receive = "cross-signed-tofu";
                    send = "cross-signed-tofu";
                    share = "cross-signed-tofu";
                  };
                };

```

The delete keys are recommended for additional security https://docs.mau.fi/bridges/general/end-to-bridge-encryption.html

Reply:
That looks great to me, it is similar to what I ended up with (except that I was too lazy to set `delete_keys.* = true`). I would also set `encrypted.require = true`. I have been using this and it looks good. It also has a very obvious failure mode

Reply:
Encryption is now default, required, the verification levels and delete keys are configured according to mautrix documentation for additional security